packages/order/src/migrations/Migration20240219102530.ts

```diff
@@ -0,0 +1,462 @@
+
+      CREATE UNIQUE INDEX IF NOT EXISTS "IDX_order_detail_order_id_item_id_version" ON "order_detail" (
+          order_id,
+          item_id,
+          version
+      );
+
+      CREATE TABLE IF NOT EXISTS "order_line_item" (
+          "id" TEXT NOT NULL,
+          "totals_id" TEXT NULL,
+          "title" TEXT NOT NULL,
+          "subtitle" TEXT NULL,
+          "thumbnail" TEXT NULL,
+          "variant_id" TEXT NULL,
+          "product_id" TEXT NULL,
+          "product_title" TEXT NULL,
+          "product_description" TEXT NULL,
+          "product_subtitle" TEXT NULL,
+          "product_type" TEXT NULL,
+          "product_collection" TEXT NULL,
+          "product_handle" TEXT NULL,
+          "variant_sku" TEXT NULL,
+          "variant_barcode" TEXT NULL,
+          "variant_title" TEXT NULL,
+          "variant_option_values" JSONB NULL,
+          "requires_shipping" BOOLEAN NOT NULL DEFAULT true,
+          "is_discountable" BOOLEAN NOT NULL DEFAULT true,
+          "is_tax_inclusive" BOOLEAN NOT NULL DEFAULT false,
+          "compare_at_unit_price" NUMERIC NULL,
+          "raw_compare_at_unit_price" JSONB NULL,
+          "unit_price" NUMERIC NOT NULL,
+          "raw_unit_price" JSONB NOT NULL,
+          "created_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "updated_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          CONSTRAINT "order_line_item_pkey" PRIMARY KEY ("id")
+      );
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_line_item_variant_id" ON "order_line_item" (
+          variant_id
+      );
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_line_item_product_id" ON "order_line_item" (
+          product_id
+      );
+
+      CREATE TABLE IF NOT EXISTS "order_line_item_tax_line" (
+          "id" TEXT NOT NULL,
+          "description" TEXT NULL,
+          "tax_rate_id" TEXT NULL,
+          "code" TEXT NOT NULL,
+          "rate" NUMERIC NOT NULL,
+          "raw_rate" JSONB NOT NULL,
+          "provider_id" TEXT NULL,
+          "created_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "updated_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "item_id" TEXT NULL,
+          CONSTRAINT "order_line_item_tax_line_pkey" PRIMARY KEY ("id")
+      );
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_line_item_tax_line_item_id" ON "order_line_item_tax_line" (item_id);
+
+      CREATE TABLE IF NOT EXISTS "order_line_item_adjustment" (
+          "id" TEXT NOT NULL,
+          "description" TEXT NULL,
+          "promotion_id" TEXT NULL,
+          "code" TEXT NULL,
+          "amount" NUMERIC NOT NULL,
+          "raw_amount" JSONB NOT NULL,
+          "provider_id" TEXT NULL,
+          "created_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "updated_at" TIMESTAMPTZ NOT NULL DEFAULT Now(),
+          "item_id" TEXT NULL,
+          CONSTRAINT "order_line_item_adjustment_pkey" PRIMARY KEY ("id"),
+          CONSTRAINT order_line_item_adjustment_check CHECK (amount >= 0)
```

Comment:
Can't an adjustment be negative?

Reply:
No in this context. But as I mentioned, I prefer to keep business logic in the app layer. That will probably be removed